File: elementary/p115.py
import math

def mod_pow(x,n,mod):
	res = 1
	while (n > 0):
		# print("n1")
		# print(n)
		# print(n&1)
		if n&1:
			# print("res")
			# print(res)
			res = res * x % mod
			# print(res)
		# print("x1")
		# print(x)
		x = x * x % mod
		# print(x)
		# print("n2")
		# print(n)
		n = n >> 1
		# print(n)
	return res

n = 8
print(mod_pow(13,n,n))
for i in range(2,n):
	print(i, mod_pow(i,n,n))
	# print(i, math.pow(i,17), math.pow(i,17)%17,i%17)

File: elementary/p115.cpp
#include<stdio.h>

typedef long long ll;

ll mod_pow(ll x, ll n, ll mod){
	ll res = 1;
	while(n>0){
		if(n&1)res = res*x%mod;
		x = x*x%mod;
		n>>=1;
	}
	return res;
}

int main(void){
	ll i;
	for(i=2;i<17;i++){
		printf("%lld\n",i);
		printf("%lld\n", mod_pow(i,17,17));
	}
}